llvm/include/llvm/ADT/ArrayRef.h

```diff
@@ -59,6 +60,8 @@ namespace llvm {
     /// The number of elements.
     size_type Length = 0;
 
+    friend MutableRowView<T>;
```

Comment:
Why do we need this?

Reply:
To access private members Length and Data.